compiler-core/src/analyse.rs

```diff
@@ -392,6 +392,14 @@ impl<'a, A> ModuleAnalyzer<'a, A> {
             ..
         } = c;
         self.check_name_case(name_location, &name, Named::Constant);
+        // If the constant's name matches an unqualified import, emit a warning:
+        if environment.unqualified_imported_names.contains_key(&name) {
+            self.problems.warning(Warning::TopLevelDefinitionShadowsImport {
+            // Use the constant's location span for highlighting
+            location: c.location,
+            name: name.clone(),
+            });
+        }
```

Comment:
Can we move this and the other instance of this block below into a helper method on `self` please 🙏